### PR TITLE
Add doc for dynamic threadpool settings

### PR DESCRIPTION
**Pull request by gbbafna**
### Description
 Doc for dynamic threadpool size cluster settings 

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8587 

### Version

2.18

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gbbafna** [comment]
Requesting @ltaragi for the tech review.

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 1]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
 
+- `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the size of fixed and resizable queue threadpool size. Overrides the defaults provided in yml.
+
+- `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Controls the maximum size of  scaling threadpool. Overrides the defaults provided in yml.
+
+- `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Controls the core size of  scaling threadpool. Overrides the defaults provided in yml.
+
+### Notes for dynamically tuning Thread Pool Settings
+
+These are expert level settings, which can potentially destabilize your cluster. It is not advised on heterogeneous clusters, as it will make threadpool count homogenous on all the nodes. It is not advised to tune threadpool which are used in both data node and cluster manager as well , due to the same reason. Post tuning these settings,
```
[the remark this answers is not in the document]
Comment (gbbafna) [reply _install-and-configure/configuring-opensearch/index-settings.md]: Addressed .

**Review by ltaragi — APPROVED** [review]

**Comment by gbbafna** [comment]
@kolchfa-aws : Can anybody from the team review this ?

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 2]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
 
 - `cluster.default_number_of_replicas` (Integer): Controls the default number of replicas for indexes in the cluster. The index-level `index.number_of_replicas` setting defaults to this value if not configured. Default is `1`.
 
+- `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the size of fixed and resizable queue threadpool size. Overrides the defaults provided in yml.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/index-settings.md:84]: Do you mean something like this?
```suggestion
- `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the sizes of both the fixed and resizable queue thread pools. Overrides the defaults provided in `opensearch.yml`.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 3]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
+
+- `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Controls the maximum size of  scaling threadpool. Overrides the defaults provided in yml.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/index-settings.md:86]:
```suggestion
- `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Sets the maximum size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 4]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
+
+- `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Controls the core size of  scaling threadpool. Overrides the defaults provided in yml.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/index-settings.md:88]:
```suggestion
- `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Specifies the core size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 5]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
+
+### Notes for dynamically tuning Thread Pool Settings
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/index-settings.md:90]:
```suggestion
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 6]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
+
+These are expert level settings, which can potentially destabilize your cluster. It is not advised on clusters having different types of hardware for same roles, as it will make threadpool count equal on all of them. It is not advised to tune threadpool which are used in both data node and cluster manager as well , due to the same reason. Post tuning these settings,
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/index-settings.md:92]:
```suggestion
Before tuning thread pool settings dynamically, note that these are expert-level settings that can potentially destabilize your cluster. Modifying thread pool settings applies the same thread pool size to all nodes, so it’s not recommended for clusters with different hardware for the same roles. Similarly, avoid tuning thread pools shared by both data nodes and cluster manager nodes for the same reason.After making these changes, we recommend monitoring your cluster to ensure it remains stable and performs as expected.
{: .warning}
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 7]
```diff
@@ -81,6 +81,17 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
+We recommend monitoring the cluster to ensure it is stable and performing as expected.
```
Comment (kolchfa-aws) [inline _install-and-configure/configuring-opensearch/index-settings.md:93]:
```suggestion
```

**Review by kolchfa-aws — COMMENTED** [review]
Thank you, @gbbafna! Some wording suggestions before we move the PR to editorial review.

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 8]
```diff
@@ -81,6 +81,16 @@ OpenSearch supports the following dynamic cluster-level index settings:
...
 
+- `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the sizes of both the fixed and resizable queue thread pools. Overrides the defaults provided in `opensearch.yml`.
+
+- `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Sets the maximum size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.
+
+- `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Specifies the core size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.
+
+
+Before tuning thread pool settings dynamically, note that these are expert-level settings that can potentially destabilize your cluster. Modifying thread pool settings applies the same thread pool size to all nodes, so it's not recommended for clusters with different hardware for the same roles. Similarly, avoid tuning thread pools shared by both data nodes and cluster manager nodes for the same reason. After making these changes, we recommend monitoring your cluster to ensure it remains stable and performs as expected.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:91]:
```suggestion
Before tuning thread pool settings dynamically, note that these are expert-level settings that can potentially destabilize your cluster. Modifying thread pool settings applies the same thread pool size to all nodes, so it's not recommended for clusters with different hardware for the same roles. Similarly, avoid tuning thread pools shared by both data nodes and cluster manager nodes. After making these changes, we recommend monitoring your cluster to ensure that it remains stable and performs as expected.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @gbbafna A couple minor changes. Thanks!

**Review by kolchfa-aws — APPROVED** [review]
LGTM

**Pull request merged**